test/integration/framework/framework.go

```diff
@@ -79,3 +82,28 @@ func AfterSuite(testEnv *envtest.Environment) {
 	err := testEnv.Stop()
 	gomega.ExpectWithOffset(1, err).NotTo(gomega.HaveOccurred())
 }
+
+func DeleteCapacity(ctx context.Context, c client.Client, cap *kueue.Capacity) error {
+	if cap != nil {
+		if err := c.Delete(ctx, cap); err != nil && !apierrors.IsNotFound(err) {
+			return err
+		}
+	}
+	return nil
+}
+
+func DeleteNamespace(ctx context.Context, c client.Client, ns *corev1.Namespace) error {
+	if err := c.DeleteAllOf(ctx, &batchv1.Job{}, client.InNamespace(ns.Name)); err != nil && !apierrors.IsNotFound(err) {
+		return err
+	}
+	if err := c.DeleteAllOf(ctx, &kueue.Queue{}, client.InNamespace(ns.Name)); err != nil && !apierrors.IsNotFound(err) {
+		return err
+	}
+	if err := c.DeleteAllOf(ctx, &kueue.QueuedWorkload{}, client.InNamespace(ns.Name)); err != nil && !apierrors.IsNotFound(err) {
+		return err
+	}
+	if err := c.Delete(ctx, ns); err != nil && !apierrors.IsNotFound(err) {
```

Comment:
Do we need to take care of the finalizer?

OTOH, this at least gets a DeletionTimestamp, which prevents any new objects being created inside the namespace, which might be enough for us.

Reply:
yeah, it doesn't really matter.